go/test/endtoend/transaction/twopc/utils/utils.go

```diff
@@ -90,3 +94,79 @@ func WriteTestCommunicationFile(t *testing.T, fileName string, content string) {
 func DeleteFile(fileName string) {
 	_ = os.Remove(path.Join(os.Getenv("VTDATAROOT"), fileName))
 }
+
+// WaitForResults waits for the results of the query to be as expected.
+func WaitForResults(t *testing.T, vtParams *mysql.ConnParams, query string, resultExpected string, waitTime time.Duration) {
+	timeout := time.After(waitTime)
```

Comment:
Personal preference to always use `context.WithTimeout` and then cehck for `<- ctx.Done()`. This would also do the right thing for `mysqlConnec(ctx, ...)`.

Reply:
for this PR, I will leave it as-is. As it is concerned with resolving flaky test issues.